### ISSUE TITLE
Unable to debug on Visual Studio 2017 Mac  (Sierra 10.12.4 Beta)

### ISSUE THREAD

**Issue by AAimson** [dotnet/coreclr#10279]
When trying to debug a MVC Web project in VS 2017 Mac I get the error message: Unable to attach to CoreCLR. 

dotnet
>Microsoft .NET Core Shared Framework Host
>Version  : 1.1.0

dotnet --info
>.NET Command Line Tools (1.0.1)
>
>Product Information:
> Version:            1.0.1
> Commit SHA-1 hash:  005db40cd1
>
>Runtime Environment:
> OS Name:     Mac OS X
> OS Version:  10.12
> OS Platform: Darwin
 >RID:         osx.10.12-x64
 >Base Path:   /opt/dotnet/sdk/1.0.1

**Comment by suqram** [comment]
Got the same issue (also on Sierrs 10.12.4 beta)

**Comment by mikem8361** [comment]
This will be fixed in the next 1.0 and 1.1 service releases and has been fixed in the master branch.

**Comment by AntoineVht** [comment]
When will it be available ? Thanks

**Comment by robbpriestley** [comment]
Sierra 10.2.4 is now being pushed as an update by Apple. I installed the update this morning and now I can no longer debug in VS Code ("Unable to attach to CoreCLR."). I believe this is the same issue reported above. I installed the latest .NET Core SDK (`dotnet --version` now says `1.0.1`). Is the fix available elsewhere, or any advice please?

**Comment by mikem8361** [comment]
Here is a work around that worked for Gregg Miskelly:
1.	Download https://dotnet.myget.org/F/dotnet-core/api/v2/package/runtime.osx.10.10-x64.Microsoft.NETCore.Runtime.CoreCLR/1.1.2-servicing-25123-01
2.	Open the resulting file as a zip, and copy out runtimes/osx.10.10-x64/native/libdbgshim.dylib
3.	Open a terminal
4.	cd ~/.vscode/extensions/ms-vscode.csharp-1.8.0/.debugger
5.	cp <path-to-extracted-libdbgshim> .
As I said above, but just to repeat here, I expect to have a real fix out soon. But hopefully this will unblock folks in the mean time.

**Comment by robbpriestley** [comment]
Confirmed workaround works. Thanks for the info. When the real fix is released, will I need to "reverse" the workaround, or will installing the new software get me back to consistency?

**Comment by armartinez** [comment]
Is there a workaround for Visual Studio Mac ?

**Comment by mikem8361** [comment]
@gregg-miskelly Is there a workaround for VS for Mac?

**Comment by MacFriis** [comment]
Until a fix for the VS for Mac is up and running, I use VS for Mac, as editor, and compiler (for checking code) then I run my project locally using the command line "dotnet run", at least I can test my code, before uploading to server, debug... not so much

**Comment by gregg-miskelly** [comment]
I don't have precise steps, but you can work around this by downloading https://vsdebugger.azureedge.net/coreclr-debug-1-9-0/coreclr-debug-osx.10.11-x64.zip, and replacing the version of vsdbg that VS for Mac is using with that version.

@DavidKarlas do you have better instructions?

**Comment by suqram** [comment]
This fixed it for me, in Visual Studio for Mac:

1. Download https://dotnet.myget.org/F/dotnet-core/api/v2/package/runtime.osx.10.10-x64.Microsoft.NETCore.Runtime.CoreCLR/1.1.2-servicing-25123-01
2. Open the resulting file as a zip, and copy out runtimes/osx.10.10-x64/native/libdbgshim.dylib
3. Navigate to /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/DotNetCore.Debugger/Adapter/ in Finder or Terminal
4. Rename libdbgshim.dylib to libdbgshim.dylib.old
5. Paste in the new libdbgshim.dylib

**Comment by MacFriis** [comment]
@suqram  You just saved my day and night! Thx.

**Comment by EltonAlvess** [comment]
@suqram it worked well!

Thanks

**Comment by swilliams-a3digital** [comment]
fix it for me!

**Comment by rosascarlos** [comment]
@suqram Excellent!
Work for me!
Thanks

**Comment by robvdveer** [comment]
Excellent, copying the libdbgshim file as instructed by @suqram did the trick.

**Comment by jhallam3** [comment]
@suqram Excellent!
Work for me!
Thanks

**Comment by kinliang** [comment]
Works for me too!

thanks!

**Comment by githubBrandon** [comment]
> Open the resulting file as a zip, and copy out runtimes/osx.10.10-x64/native/libdbgshim.dylib

@mikem8361 what does this mean?

**Comment by mikem8361** [comment]
A nuget file is a zip file. Rename to .zip and run unzip on it.

**Comment by bryanross** [comment]
@suqram Work for me as well - thank you.

**Comment by RexLKW** [comment]
@suqram Thanks so much! It works!

**Comment by robbpriestley** [comment]
Is this issue fixed in production now, or is it still in workaround? I noticed a new C# Extension and installed it yesterday. Does that contain the fix?

**Comment by LokeSonne** [comment]
@robbpriestley Sure seems like it. This is in the changelog for the extension in vs code:
What's New in 1.8.1
Fixes debugging on macOS Sierra 10.12.4.

(and it's working)
Brilliant!

**Comment by joristt** [comment]
I did precisely what @suqram suggested, but now I get a popup when I try to run my application:

> Debugger operation failed
> Unknown Error: 0x92330062

I don't get the "Unable to attach to CoreCLR" message in my Application Output window anymore, so that's fixed.

**Comment by gregg-miskelly** [comment]
@joristt if you have the 1.8.1 extension installed and you are still running into an issue, your issue isn't related to this issue. Please open a new issue, this issue is long enough :).

**Comment by robvdveer** [comment]
I was working on aspnet core when this issue started, and with VS preview it used to start a new Browser with a new debug session. Am i looking for the wrong setting now?

**Comment by joristt** [comment]
@gregg-miskelly Yeah you're right, sorry :). You said
> replace the version of vsdbg that VS for Mac is using

and I interpreted that as: replace only the vsdbg Unix executable. Tried copying everything from that zip to the VS Adapter directory 1 minute after commenting here and that fixed the problem.

**Comment by gregg-miskelly** [comment]
@robvdveer by 'VS Preview' you mean 'VS For Mac Preview' correct? If so - VS For Mac hasn't shipped an update vsdbg package containing the fix yet (at least to my knowledge), so you are very likely at the right place. There are two work arounds posted above, but I will give you the latest version of it --

* Download https://vsdebugger.azureedge.net/coreclr-debug-1-9-1/coreclr-debug-osx.10.11-x64.zip
* Navigate to /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/DotNetCore.Debugger/Adapter/ in Finder or Terminal
* Extract the zip to this folder

**Comment by robvdveer** [comment]
Thanks @gregg-miskelly for such a quick response!

**Comment by knilecrack** [comment]
@gregg-miskelly  well played. Thanks

**Comment by akhilanair94** [comment]
@gregg-miskelly saved the day!!!!

**Comment by mclarenengineering** [comment]
@gregg-miskelly this will work on the visual studio preview the debug gives an execution error
